htlcswitch/switch.go

```diff
@@ -524,7 +528,12 @@ func (s *Switch) GetAttemptResult(attemptID uint64, paymentHash lntypes.Hash,
 			return
 		}
 		resultChan <- result
-	}()
+	})
+
+	// The switch shutting down is signaled by closing the channel.
+	if err != nil {
```

Comment:
I think the switch shutting down and an error from the goroutine manager are different?

Reply:
GoroutineManager can only return an error, if it is stopping. I added a check just in case:

```go
        // The switch shutting down is signaled by closing the channel.
        if errors.Is(err, fn.ErrStopping) {
                close(resultChan)
        } else if err != nil {
                return nil, fmt.Errorf("got an unexpected error from "+
                        "GoroutineManager: %w", err)
        }
```